Jellyfin.Plugin.PlaybackReporting/Data/ActivityRepository.cs

```diff
@@ -413,7 +413,12 @@ public void UpdatePlaybackAction(PlaybackInfo playInfo)
             {
                 using var connection = CreateConnection(true);
                 using var statement = connection.PrepareStatement(sql_query);
-                var fromDate = DateTime.Parse(date + " 00:00:00").AddHours(-timezoneOffset);
+                // Convert the decimal to minutes
+                // 0.5*60 = 30
+                int minOffset = (int)(( timezoneOffset < 0.0 ? (-timezoneOffset)%1 : timezoneOffset%1 )*60);
+                // Turn to negative in case timezoneOffset is negative
+                minOffset = ( timezoneOffset < 0.0 ? -minOffset : minOffset );
+                var fromDate = DateTime.Parse(date + " 00:00:00").AddHours((int)-timezoneOffset).AddMinutes(-minOffset);
```

Comment:
if you extract the other 6 occurrences into a func and craft it right you could also use it here I would wager

Reply:
Could you take a look now?